packages/build-tools/src/builders/android.ts

```diff
@@ -74,7 +76,7 @@ async function buildAsync(ctx: BuildContext<Android.Job>): Promise<void> {
   });
 
   await ctx.runBuildPhase(BuildPhase.RESTORE_CACHE, async () => {
-    if (ctx.env.EAS_USE_CACHE !== '1') {
+    if (ctx.env.EAS_USE_CACHE !== '1' && ctx.env.EAS_RESTORE_CACHE !== '1') {
```

Comment:
Actually there's one more change we should do: run `await ctx.cacheManager?.restoreCache(ctx);` always (we never said we're turning off the old cache).

Also, can we please rewrite the logic to something more verbose, but more easily understandable? What I mean is that
```
  (USE_CACHE == 1 && RESTORE_CACHE != 0) // we want to use cache, but are forbidden from restoring
  ||
  (RESTORE_CACHE == 1) // we want to restore no matter what
```
is more readable to me as a human than two AND-ed negations. Are these two equivalent?